src/main/java/com/mymemor/mymemor/controller/SearchController.java

```diff
@@ -1,16 +1,82 @@
 package com.mymemor.mymemor.controller;
 
-import org.springframework.web.bind.annotation.PostMapping;
+import com.mymemor.mymemor.Utils;
+import com.mymemor.mymemor.exceptions.EmptyQuery;
+import com.mymemor.mymemor.exceptions.InvalidName;
+import com.mymemor.mymemor.exceptions.InvalidUserName;
+import com.mymemor.mymemor.model.User;
+import com.mymemor.mymemor.repository.AccountRepository;
+import com.mymemor.mymemor.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
 @RestController
 @RequestMapping("/api")
 public class SearchController {
 
-    @PostMapping("/search")
-    public String getSearchResultViaAjax() {
-        return "";
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private AccountRepository accountRepository;
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    private List<User> getUsersByIds(List ids){
+        List<User> listUser = new ArrayList<>();
+        for(Object id: ids){
+            User user = userRepository.findById(((BigInteger) id).longValue()).orElseThrow();
+            listUser.add(user);
+        }
+        return listUser;
+    }
+
+    private List getUserIdsByUsername(EntityManager entityManager, String query){
```

Comment:
Update function name and return type.
```suggestion
    private List<User> getUsersByUsername(EntityManager entityManager, String query){
```